File: TrelloApp/src/app/core/models/list.ts
import { Card } from "./card"
import {Board} from './board';

export interface List {
  id: number,
  title: string,
  position: number,
  boardId: number,
  createdAt: string,
  updatedAt?: string,
  board: Board,
  cards: Card[]
}

export interface AddList {
  title: string,
  position: number
}

export interface UpdateList {
  title?: string,
  position?: number
}

File: TrelloApi/TrelloApi/src/Domain/Entities/List.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TrelloApi.Domain.Entities;

[Table("List")]
public class List: Entity
{
    [StringLength(32), Required]
    public string Title { get; set; }
    
    public int Position { get; set; }
    
    [ForeignKey("Board"), Required]
    public int BoardId { get; set; }
    public Board Board { get; set; }
    
    public ICollection<Card> Cards { get; set; }
    
    public List(string title, int boardId, int position)
    {
        Title = title;
        BoardId = boardId;
        Position = position;
        CreatedAt = DateTime.UtcNow;
        UpdatedAt = null;
    }
}
